web/src/api/getState.ts

```diff
@@ -2,14 +2,33 @@ import type { UseQueryResult } from '@tanstack/react-query';
 import { useQuery } from '@tanstack/react-query';
 import { useAtom, useAtomValue } from 'jotai';
 import type { GridState } from 'types';
-import { isHourlyAtom, timeAverageAtom } from 'utils/state/atoms';
+import { hasPathTimeAverageAndDatetime, parsePath } from 'utils/pathUtils';
+import {
+  isHourlyAtom,
+  targetDatetimeStringAtom,
+  timeAverageAtom,
+} from 'utils/state/atoms';
 
-import { cacheBuster, getBasePath, QUERY_KEYS } from './helpers';
+import { cacheBuster, getBasePath, isValidDate, QUERY_KEYS } from './helpers';
 
-const getState = async (timeAverage: string): Promise<GridState> => {
-  const path: URL = new URL(`v8/state/${timeAverage}`, getBasePath());
-  path.searchParams.append('cacheKey', cacheBuster());
+const getState = async (
+  timeAverage: string,
+  targetDatetime?: string
+): Promise<GridState> => {
+  const parsedPath = parsePath(location.pathname);
+  const isValidDatetime = targetDatetime && isValidDate(targetDatetime);
+  const timeAverageToQuery = parsedPath?.timeAverage || timeAverage;
+  const path: URL = new URL(
+    `v8/state/${timeAverageToQuery}${
+      isValidDatetime ? `?targetDate=${targetDatetime}` : ''
+    }`,
+    getBasePath()
+  );
 
+  path.searchParams.append(
+    'cacheKey',
+    isValidDatetime && targetDatetime ? targetDatetime : cacheBuster()
+  );
```

Comment:
I think we can skip adding this entriely if the targetDatetime is valid.

We already have the targetDate in the query parameters that will bust the cache.

Reply:
Good point, it was useful to see for testing while building. I'll try remove it now